index.py

```diff
@@ -105,8 +105,12 @@ def build(self):
             if i == 0 or term != tokens_list[i-1][0]:
                 # new term
                 self.dictionary[term] = PostingList()
-            # insert into appropriate PostingList
-            self.dictionary[term].insert(curr_tuple[0], curr_tuple[1], curr_tuple[2])
+            # Insert into appropriate PostingList
+            # if same term and docID, do not increment PL.size
+            if term == tokens_list[i-1][0] and curr_tuple[0] == tokens_list[i-1][1][0]:
```

Comment:
Might have an error here if `i` is 0 cos `tokens_list[-1]`

Reply:
Might want to check if `i==0`

Reply:
ok